lib/DocumentManager.php

```diff
@@ -150,13 +200,37 @@ public function createQuery($query, $locale = null, array $options = [])
      */
     public function createQueryBuilder()
     {
-        $event = new Event\QueryCreateBuilderEvent();
-        $this->eventDispatcher->dispatch(Events::QUERY_CREATE_BUILDER, $event);
+        $event = new Event\QueryCreateBuilderEvent($this->context);
+        $this->dispatch(Events::QUERY_CREATE_BUILDER, $event);
 
         return $event->getQueryBuilder();
     }
 
     /**
+     * TODO : This interface method.
+     * {@inheritdoc}
+     */
+    public function getInspector()
+    {
+        return $this->inspectorFactory->getInspector($this->context);
+    }
+
+    public function getNodeManager()
+    {
+        return $this->nodeManager;
+    }
+
+    public function getRegistry()
+    {
+        return $this->registry;
+    }
+
+    public function getProxyFactory()
```

Comment:
These methods (`getNodeManager`, `getRegistry` and `getProxyFactory`) are required by the legacy Sulu system which is not using events. The API should NOT be exposing these methods, but not currently sure of a better way to handle this.